src/backendMiddleware.ts

```diff
@@ -44,14 +43,22 @@ export class BackendMiddleware {
     await this.storageLib.initConnection()
   }
 
-  public async setIdentityWallet(
+  public setIdentityWallet(identityWallet: IdentityWallet) {
```

Comment:
👍

Regarding the second function, `authenticateAndSetIdentityWallet`, I am debating the naming, since it should really be something like `authenticateAndCacheAndSetIdentityWallet` :D

Reply:
Addressed in the latest commit.